src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx

```diff
@@ -26,6 +26,9 @@ const SubjectsStep = ({ stepLabel, btnsBox }) => {
   })
 
   const [subjectError, setSubjectError] = useState('')
+  const [subjectFetched, setSubjectIsFetched] = useState(false)
```

Comment:
Can you remove this state and getSubjects when subject.category not falsy:
const getSubjectsNames = useCallback(() => {
if (subjects.category) {
    return subjectService.getSubjectsNames(subjects.category._id)
}
}, [subjects.category] )

Reply:
it will be a type error because service func should return promise, not undefined, also without state it will fetch on every focus I think